api/routes.go

```diff
@@ -65,17 +80,26 @@ func (api *API) noValidate(h httprouter.Handle) httprouter.Handle {
 func (api *API) validate(h httprouter.Handle) httprouter.Handle {
 	return func(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
 		api.logRequest(req)
-		tokenStr, err := tokenFromRequest(req)
-		if err != nil {
-			api.staticLogger.Debugln("Error fetching token from request:", err)
-			api.WriteError(w, err, http.StatusUnauthorized)
-			return
-		}
-		token, err := jwt.ValidateToken(tokenStr)
-		if err != nil {
-			api.staticLogger.Debugln("Error validating token:", err)
-			api.WriteError(w, err, http.StatusUnauthorized)
-			return
+		var token jwt2.Token
```

Comment:
maybe it makes sense to rename this function to `WithAuthentication`, to be consistent with `WithDBSession` and also to be more descriptive, I think `validate` is not a good name for this middleware